File: app/src/main/java/com/mehmetakiftutuncu/muezzin/adapters/viewholders/CountryViewHolder.java
package com.mehmetakiftutuncu.muezzin.adapters.viewholders;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.mehmetakiftutuncu.muezzin.R;
import com.mehmetakiftutuncu.muezzin.fragments.CountrySelectionFragment;
import com.mehmetakiftutuncu.muezzin.models.Country;

/**
 * Created by akif on 11/05/16.
 */
public class CountryViewHolder extends RecyclerView.ViewHolder {
    private Context context;

    private View countryItemLayout;
    private TextView textViewName;
    private TextView textViewNativeName;

    private CountrySelectionFragment.OnCountrySelectedListener onCountrySelectedListener;

    public CountryViewHolder(Context context, View countryItemLayout, CountrySelectionFragment.OnCountrySelectedListener onCountrySelectedListener) {
        super(countryItemLayout);

        this.context = context;
        this.countryItemLayout = countryItemLayout;
        this.onCountrySelectedListener = onCountrySelectedListener;

        textViewName       = (TextView) countryItemLayout.findViewById(R.id.textView_item_country_name);
        textViewNativeName = (TextView) countryItemLayout.findViewById(R.id.textView_item_country_nativeName);
    }

    public void setFrom(final Country country) {
        String nameToUse = country.getLocalizedName(context);

        textViewName.setText(nameToUse);
        textViewNativeName.setText(country.nameNative);

        countryItemLayout.setOnClickListener(v -> onCountrySelectedListener.onCountrySelected(country));
    }
}

File: app/src/main/java/com/mehmetakiftutuncu/muezzin/adapters/viewholders/CountryViewHolder.kt
package com.mehmetakiftutuncu.muezzin.adapters.viewholders

import android.content.Context
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mehmetakiftutuncu.muezzin.R
import com.mehmetakiftutuncu.muezzin.fragments.SelectionFragment
import com.mehmetakiftutuncu.muezzin.models.Country

class CountryViewHolder(private val ctx: Context,
                        private val view: View,
                        private val listener: SelectionFragment.OnSelectedListener<Country>): RecyclerView.ViewHolder(view) {
    private val textViewName: TextView = view.findViewById(R.id.textView_item_country_name)
    private val textViewNativeName: TextView = view.findViewById(R.id.textView_item_country_nativeName)

    fun set(country: Country) {
        textViewName.text = country.localizedName(ctx)
        textViewNativeName.text = country.nameNative

        view.setOnClickListener {
            listener.onSelected(country)
        }
    }
}